src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +87,12 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex().equals(Person.Sex.WOMAN) && p.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
+                .map(Cat::getName)
```

Comment:
Good, I like that style :) It's an appreciation comment, don't fix it)
```suggestion
                .map(Person::getCats)
                .flatMap(Collection::stream)
                .map(Cat::getName)
```

Reply:
Yes, it looks pretty beauty